File: verifier-cli/src/main/java/com/jetbrains/pluginverifier/verifiers/VerificationContext.java
package com.jetbrains.pluginverifier.verifiers;

import com.intellij.structure.domain.Ide;
import com.intellij.structure.domain.Plugin;
import com.intellij.structure.resolvers.Resolver;
import com.jetbrains.pluginverifier.api.VOptions;
import com.jetbrains.pluginverifier.location.ProblemLocation;
import com.jetbrains.pluginverifier.problems.Problem;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author Sergey Evdokimov
 */
public interface VerificationContext {
  @NotNull
  VOptions getVerifierOptions();

  void registerProblem(@NotNull Problem problem, @NotNull ProblemLocation location);

  @NotNull
  String getOverview();

  @NotNull
  Plugin getPlugin();

  @NotNull
  Ide getIde();

  @NotNull
  Resolver getIdeResolver();

  @NotNull
  Resolver getJdkResolver();

  @Nullable
  Resolver getExternalClassPath();
}

File: verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/VerificationContext.kt
package com.jetbrains.pluginverifier.verifiers

import com.jetbrains.pluginverifier.verifiers.resolution.ClassResolver

interface VerificationContext {
  val classResolver: ClassResolver

  val problemRegistrar: ProblemRegistrar

  val deprecatedApiRegistrar: DeprecatedApiRegistrar

  val experimentalApiRegistrar: ExperimentalApiRegistrar
}